src/scikit_build_core/builder/builder.py

```diff
@@ -84,6 +86,68 @@ def _filter_env_cmake_args(env_cmake_args: list[str]) -> Generator[str, None, No
             yield arg
 
 
+# Type-hinting for Traversable is rather hard because they were introduced in python 3.11.
+# This avoids introducing importlib_resources dependency that may not be used in the actual package loader
+def _sanitize_path(path: Any) -> list[Path] | None:
+    if isinstance(path, MultiplexedPath):
+        # pylint: disable-next=protected-access
+        return path._paths
```

Comment:
I'm not fond of using a hidden member here. How about using the modern Python's support for directories, and just using the hidden member on older Pythons?

Reply:
> How about using the modern Python's support for directories

I am not sure I get this part. The `MultiplexedPath` can point to `/rootA/base` + `/rootB/base` and we need the to point to all possible `base` folders to pass it to CMake. The only support that I know of would be `joinpath` like support, but that wouldn't help us here. Maybe we could run `iterdir` without recursion 🤔

Reply:
Modern Python (3.12) can do:

```python
with importlib.resources.as_file(some_directory) as d:
   # d is a real directory
```

I believe this combines all the directories into a real directory for the duration of the block.

Reply:
But we need to pass these as `str` to build the CMakeCacheInit. When I checked the implementation of `MultiplexedPath`, whenever it transformed to a `pathlib.Path` and it was pointing to a directory-like path, it would take the first value of `_paths`, so that would not help here.

One thing that would work is to `iterdir` until we hit `pathlib.Path` and step back from there, but we would need to do that until all `MultiplexedPath` are resolved in order to make sure all possible paths are resolved.

Reply:
It is guaranteed to be a single real path to a real directory inside the context manager. You can get a string to it. I believe it copies the merged directory contents to a temporary directory. After the context manager, the temporary directory is removed. The context manager just has to live for the duration of the build.

Reply:
Ok, I checked and indeed that is the case. I'm trying to think if there are some breakages because of that, but nothing related to the current implementation thus far. It could get wonky if we support editable installs pointing to the build directory, but there would be many other weirdness there.

If we use context for this, how do we keep the context alive for all the entry-points during the build-install-archive? E.g. the rpaths could use `CMAKE_INSTALL_RPATH_USE_LINK_PATH` and we should keep it resolvable until auditwheel can patch those

Reply:
Okay, I think we can't avoid using `_paths` then.